imports/string/shared.lua

```diff
@@ -0,0 +1,52 @@
+---@class oxstring : stringlib
+lib.string = string
+
+local StringCharset = {}
+local NumberCharset = {}
+
+for i = 48, 57 do NumberCharset[#NumberCharset + 1] = string.char(i) end
+for i = 65, 90 do StringCharset[#StringCharset + 1] = string.char(i) end
+for i = 97, 122 do StringCharset[#StringCharset + 1] = string.char(i) end
+
+---Takes a number of characters and returns a random string.
+---@param length number
+---@return string
+function string.randomString(length)
+    if length <= 0 then return '' end
+    return string.randomString(length - 1) .. StringCharset[math.random(1, #StringCharset)]
+end
+
+---Takes a string and a delimiter and returns the splitted string.
+---@param input string
+---@param delimiter string
+---@return table
+function string.splitString(input, delimiter)
+    local result = {}
+    local from = 1
+    local delim_from, delim_to = string.find(input, delimiter, from)
```

Comment:
You can use gmatch instead of find to get an iterator instead of tracking these pointers.